src/js/actions/libraries.js

```diff
@@ -668,36 +672,45 @@ define(function (require, exports) {
             }
         }
         
-        this.dispatch(events.libraries.UPDATING_GRAPHIC_CONTENT, { documentID: documentID, element: element });
-
-        library.beginOperation();
+        var document = this.flux.stores.document.getDocument(documentID),
+            updateGraphicPromise = Promise.resolve();
         
-        // If the element we're editing was deleted, we recreate it as a new element, so the changes aren't lost.
-        // Otherwise, we just remove the existing representations so we can add the new ones
-        if (element.deletedLocally) {
-            element = library.createElement(element.name, element.type);
-        } else {
-            // TODO should keep representation's stock data
-            element.removeAllRepresentations();
+        if (!document.unsupported) {
+            this.dispatch(events.libraries.UPDATING_GRAPHIC_CONTENT, { documentID: documentID, element: element });
```

Comment:
Can you expand on the reason of the rule? I don't see the difference of invoking `dispatch` alone or follow with a promise chain, as it is running synchronously. Even if `dispatch` throws an error, it will be caught by the outer promise chain wrapped by the fluxcontroller, isn't it?